### ISSUE TITLE
🐛 [Bug Report] Possible to pick up BG flag while stealthed

### ISSUE THREAD

**Issue by henhouse** [cmangos/issues#3063]
### Bug Details

When sitting and picking up a BG flag, it is possible to pick up the flag while stealthed, as long as you are sitting.

If you are sitting (don't have to be stealthed) you receive a "You must be standing to do that" error, but the spell is still applied to the player. Perhaps this has some relation to the bug.

https://user-images.githubusercontent.com/30731/178121323-998b8ec4-c02c-4e8f-863f-42b71ba582f3.mp4

### Steps to Reproduce

1. Pick up the flag in a BG
2. Drop the flag (right-click on buff)
3. Enter stealth
4. Sit
5. Click on flag to pick up again

### Expected behavior

It either should not be possible to pick up the flag while sitting, or it should at least break stealth. I'm not sure which is  correct.

### Core SHA1 Commit Hash

852785df6

### Database SHA1 Commit Hash

d268ff1053f10771fe0d1f5a1207e5b81028d5ea

### Operating System

Linux (Ubuntu 20.04 LTS)

### Client Version

2.4.3 (The Burning Crusade)

**Comment by miraco** [comment]
I think if you click on the flag the player should stand up and remove stealth.

Maybe nood a good solution but you could add:
`if(player->HasAuraType(SPELL_AURA_MOD_STEALTH))
        player->RemoveSpellsCausingAura(SPELL_AURA_MOD_STEALTH);`
        
To HandlePlayerClickedOnFlag in BattleGroundWS.cpp

**Comment by henhouse** [comment]
> I think if you click on the flag the player should stand up and remove stealth.
> 
> Maybe nood a good solution but you could add: `if(player->HasAuraType(SPELL_AURA_MOD_STEALTH)) player->RemoveSpellsCausingAura(SPELL_AURA_MOD_STEALTH);`
> 
> To HandlePlayerClickedOnFlag in BattleGroundWS.cpp

Thanks! Since we think players should stand (since they usually do when interacting with a gameobject), I tested making the player just stand, and that fixes the issue. Sort of a hackfix (since I think it should be solved in GO interaction elsewhere), but in case anyone needs a quick fix for it, here it is:

```diff
diff --git a/src/game/BattleGround/BattleGroundEY.cpp b/src/game/BattleGround/BattleGroundEY.cpp
index 1a28f05cb..80f921dc8 100644
--- a/src/game/BattleGround/BattleGroundEY.cpp
+++ b/src/game/BattleGround/BattleGroundEY.cpp
@@ -471,6 +471,9 @@ void BattleGroundEY::HandlePlayerClickedOnFlag(Player* source, GameObject* go)
     if (GetStatus() != STATUS_IN_PROGRESS || IsFlagPickedUp() || !source->IsWithinDistInMap(go, 10))
         return;
 
+    if (source->IsSitState())
+        source->SetStandState(UNIT_STAND_STATE_STAND);
+
     DEBUG_LOG("BattleGroundEY: Team %u has taken the flag, gameobject entry %u.", source->GetTeam(), go->GetEntry());
 
     // remove world state for base flag
diff --git a/src/game/BattleGround/BattleGroundWS.cpp b/src/game/BattleGround/BattleGroundWS.cpp
index d8d4989ee..e20881666 100644
--- a/src/game/BattleGround/BattleGroundWS.cpp
+++ b/src/game/BattleGround/BattleGroundWS.cpp
@@ -320,6 +320,9 @@ void BattleGroundWS::ProcessFlagPickUpFromBase(Player* player, Team attackerTeam
     if (m_flagState[otherTeamIdx] != BG_WS_FLAG_STATE_ON_BASE)
         return;
 
+    if (player->IsSitState())
+        player->SetStandState(UNIT_STAND_STATE_STAND);
+
     DEBUG_LOG("BattleGroundWS: Team %u has taken the enemy flag.", attackerTeam);
 
     SpawnEvent(otherTeamIdx, 0, false);
@@ -383,6 +386,9 @@ void BattleGroundWS::ProcessDroppedFlagActions(Player* player, GameObject* targe
         SpawnEvent(otherTeamIdx, 0, false);
         SetFlagCarrier(otherTeamIdx, player->GetObjectGuid());
 
+        if (player->IsSitState())
+            player->SetStandState(UNIT_STAND_STATE_STAND);
+
         player->CastSpell(player, wsgFlagData[otherTeamIdx][actionId].spellId, TRIGGERED_OLD_TRIGGERED);
         if (m_brutalAssaultActive)
             player->CastSpell(player, BG_WS_SPELL_BRUTAL_ASSAULT, TRIGGERED_OLD_TRIGGERED);

```

**Comment by killerwife** [comment]
![image](https://user-images.githubusercontent.com/7995382/178138594-fd0b6e0e-0e28-4242-863a-fbc77c2a842c.png)

**Comment by killerwife** [comment]
Try that.

**Comment by henhouse** [comment]
@killerwife sadly no difference. Key issue seems to be the "You cannot do that while sitting" since forcing the player to stand in my hackfix resolves the issue. I'm guessing GO interaction should make players stand normally?

Also worth noting the EotS flag does NOT have any of the `_WHILE_SITTING` atrributes.

**Comment by killerwife** [comment]
This is not about standing up, this is about stealth not proccing while sitting.

**Comment by henhouse** [comment]
This was fixed in e82f26324d67c85cab53f09bf2a4b7a25caa802b.
